_posts/2023-03-22-#27050.md

```diff
@@ -10,15 +10,59 @@ status: upcoming
--->
+- [BIP 141
+  (SegWit)](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki)
+  introduced a new structure called a `witness` that is committed to blocks
+  separately from the transaction merkle tree. Witnesses contain data required
+  to check transaction validity but not required to determine transaction
+  effects (output consumption/creation). In other words, witnesses are used to
+  validate the blockchain state not to determine what that state is.
+
+  - Witnesses are commited to by placing the root of the witness merkle tree in
+    the block's coinbase transaction. By doing so, the witnesses are commited to
+    in the transaction merkle tree through the coinbase transaction. Nesting the
+    witness commitment in the coinbase transaction was done to make SegWit
+    soft-fork compatible.
+
+- `Assume-valid` is a node setting that makes the node skip transaction
+  validity checks (signature and script checks) prior to a pre-determined
+  "known to be good" block (assume-valid point). The default assume-valid point
+  is set by the developers and is updated every release but users have the
+  ability to set their own assume-valid point through the `-assumevalid`
+  setting.
 
+  - The assume-valid feature does not significanlty change Bitcoin's security
+    assumptions. If developers (and everyone reviewing the code changes) were
+    to conspire with miners to build a more-work chain with invalid signatures
+    (and go undetected for weeks), and then include it as the default
+    assume-valid point, they could get the network to accept an invalid chain.
+    However, those same people already have that power by just changing the
+    code - which would be much less obvious.
+    ([quoted](https://bitcoin.stackexchange.com/questions/59940/what-are-the-trust-assumptions-in-assumed-valid-in-bitcoin-core-0-14))
+    Additionally, as long as the full chain history remains available for
+    auditing it would be hard for such an attack to go unnoticed.
+
+  - It is also important to note that the configured assume-valid point does not
+    dictate which chain a node follows. Meaning that a large reorg would be able
+    to orphan (parts of) the assumed-valid chain.
+
+- Nodes in prune mode (enabled by the `-prune` setting) fully download and
+  validate the chain history but only store a (configurable) portion of the
+  recent history.
+
+- [PR 27050](https://github.com/bitcoin/bitcoin/pull/27050) proposes to skip
+  downloading the witnesses for blocks prior to the configured assume-valid
+  point, for nodes running in prune mode. The rational for this change is that
+  pruned nodes currently download witnesses but then (prior to the assume-valid
+  point) don't validate them and delete them shortly after. So why not skip
+  downloading those witnesses and save some bandwidth?
```

Comment:
maybe mention how much bandwidth because "some" kind of undersells it

Reply:
Was thinking of asking the participants how much bandwidth it saves in the questions. Is that to much homework?